### PR TITLE
fix(ethflow): prevent ethflow on wrong network

### PR DESCRIPTION
**Pull request by alfetopito**
# Summary

Recently, there was a user who sent mainnet ETH to a mainnet contract corresponding to the ETH FLOW contract on arb1.

We have not been able to reproduce this internally so far, so I don't expect this change to be really testable.
Still, it prevents this from ever happening, by doing a last minute check whether the contract matches the chain, before the ETH FLOW tx is sent.

# To Test

1. Place eth flow txs on all chains
* Should work as expected

### CODE REVIEW

File: apps/cowswap-frontend/src/modules/swap/services/ethFlow/index.ts [hunk 1]
```diff
@@ -78,6 +82,10 @@ export async function ethFlow(
       throw new Error('Quote expired. Please refresh.')
     }
 
+    if (contract.address !== getEthFlowContractAddress(chainId)) {
+      throw new Error('EthFlow contract address mismatch. Please refresh the page and try again.')
+    }
```
Comment (alfetopito) [inline apps/cowswap-frontend/src/modules/swap/services/ethFlow/index.ts:87]: This is the main change

If somehow the contract address doesn't match the one that should be used for the chain, prevent user from continuing.

**Review by shoom3301 — APPROVED** [review]

File: apps/cowswap-frontend/src/modules/swap/services/ethFlow/index.ts (hunk 1 above)
Comment (shoom3301) [reply apps/cowswap-frontend/src/modules/swap/services/ethFlow/index.ts:87]: What if chainId is wrong here?
I mean, user is on Mainnet but for some reason contract comes from Arbitrum and `chainId` here is also Arbitrum

Comment (anxolin) [reply apps/cowswap-frontend/src/modules/swap/services/ethFlow/index.ts:87]: Maybe you can make sure inside the method `ethFlowContract.createOrder`, you can make a check of what is the `chainId` of `ethFlowContract.provider`?

Comment (alfetopito) [reply apps/cowswap-frontend/src/modules/swap/services/ethFlow/index.ts:87]: > What if chainId is wrong here?
Well, good point.

I'll try Anxo's idea to solve this.

**Review by elena-zh — APPROVED** [review]

File: apps/cowswap-frontend/src/modules/swap/services/ethFlow/index.ts (hunk 1 above)
Comment (alfetopito) [reply apps/cowswap-frontend/src/modules/swap/services/ethFlow/index.ts:87]: @anxolin @shoom3301 added another conditional which checks for the current chain from the wallet https://github.com/cowprotocol/cowswap/pull/4629/commits/5c794167f544bb950d5c4db0e956836ec222fd24

**Review by shoom3301 — APPROVED** [review]

**Review by anxolin — APPROVED** [review]

**Pull request merged**